### ISSUE TITLE
Enforce multiline lambda parameters to be on the first line

### ISSUE THREAD

**Issue by t3hk0d3** [pinterest/ktlint#2121]
## Expected Rule behavior

Currently it is possible to have lambdas to be formatted like following:

```kotlin
foo.bar { 
    param -> 
        param.doSomething()
         // ...
}
```

However this is incorrect behaviour according to Kotlin Code Conventions:

https://kotlinlang.org/docs/coding-conventions.html#lambdas

> When declaring parameter names in a multiline lambda, put the names on the first line, followed by the arrow and the newline:

```kotlin
appendCommaSeparated(properties) { prop ->
    val propertyValue = prop.get(obj)  // ...
}
```

## Additional information
* Current version of ktlint: 0.49.1
* Styleguide section: https://kotlinlang.org/docs/coding-conventions.html#lambdas

**Comment by paul-dingemans** [comment]
> Currently it is possible to have lambdas to be formatted like following:
> 
> ```kotlin
> foo.bar { 
>     param -> 
>         param.doSomething()
>          // ...
> }
> ```

Code above is *not* accepted by `ktlint`. With code style `intellij_idea` or `android_studio`, this code is formatted as:
```
fun foo2() {
    foo1.bar {
            param ->
        param.doSomething()
    }
}
```
which indeed deviates from the kotlin coding conventions. 

The idea behind this deviation is that code sample like below:
```
fun foo3() {
    foo1.bar {
        param1,
        param2 ->
            param1.doSomething()
            param2.doSomething()
    }
}
```
are nicely formatted as:
```
fun foo3() {
    foo1.bar {
            param1,
            param2 ->
        param1.doSomething()
        param2.doSomething()
    }
}
```
which is more consist with indentation of parameters in a function call:
```
fun foo(
    param1: Any,
    param2: Any
)  {
    // do something
}
```

Note that with code style `ktlint_official` there is additional formatting to change code below:
```
fun foo4() {
    foo1.bar { param1,
        param2 ->
            param1.doSomething()
            param2.doSomething()
    }
}
```
to
```
fun foo4() {
    foo1.bar {
            param1,
            param2,
        ->
        param1.doSomething()
        param2.doSomething()
    }
}
```

**Comment by paul-dingemans** [comment]
I have misread the original post and focused too much on the indentation only. 

The original post identifies that code:
```
foo.bar { 
    param -> 
        param.doSomething()
         // ...
}
```
should be formatted as:
```
foo.bar {  param -> 
    param.doSomething()
    // ...
}
```
to comply with:
> When declaring parameter names in a multiline lambda, put the names on the first line, followed by the arrow and the newline:

For this a new rule is needed. This rule should also take care of:
> If the parameter list is too long to fit on a line, put the arrow on a separate line:
> ```
> foo {
>     context: Context,
>     environment: Env
>     ->
>     context.configureEnv(environment)
>. }
> ```

**Comment by paul-dingemans** [comment]
The Kotlin Coding conventions are not entirely comsistent regard indenting of a multiline parameter list.

https://kotlinlang.org/docs/coding-conventions.html#lambdas
```
foo {
   context: Context,
   environment: Env
   ->
   context.configureEnv(environment)
}
```

https://kotlinlang.org/docs/coding-conventions.html#parameters-in-lambdas
```
fun main() {
    val x = {
            x: Comparable<Number>,
            y: Iterable<Number>, // trailing comma
        ->
        println("1")
    }
    println(x)
}
```

**Comment by t3hk0d3** [comment]
Thanks for a rapid fix!
